receiver/prometheusreceiver/internal/staleness_store.go

```diff
@@ -0,0 +1,78 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"github.com/prometheus/prometheus/pkg/labels"
+)
+
+type stalenessStore struct {
+	currentHashes  map[uint64]bool
+	previousHashes map[uint64]bool
+	previous       []labels.Labels
+	current        []labels.Labels
+}
+
+func newStalenessStore() *stalenessStore {
+	return &stalenessStore{
+		previousHashes: make(map[uint64]bool),
+		currentHashes:  make(map[uint64]bool),
+	}
+}
+
+// refresh copies over all the current values to previous, and prepares.
```

Comment:
It's not clear to me whose responsibility it is to call refresh. Can you add to the method doc that refresh is called after scrape. The tests are doing a better job how to use the stalenessStore but it's not clear from the package's itself.

Reply:
Added, thank you, PTAL.